InventoryIndexer/Model/Queue/ReservationData.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Model\Queue;
+
+/**
+ * Data object for reservations queue request.
+ */
+class ReservationData
```

Comment:
do we need this class?  I mean if we are not going to expose that through the web api, and we don't want to consider that as an extension point. Why not work with an array? at least that would be faster

Reply:
Messages architecture does not support associative arrays.
In our case, we need to have information about SKU and Stock Id.
This means that next format is not supported:
`[
        'stock' => 'stoc_id',
        'sku' => 'prdocut_sku'
]`

And the only format will be supported:
`[
        0 => 'stock_id',
        1 => 'product_sku'
]`
To play with numeric array indexes looks even worse for me than DTO.